### DISCUSSION TITLE
libjpeg "no platform load command found" warnings in xCode

### DISCUSSION THREAD

**Discussion by plangworthy** [axmolengine/axmol#1338] [General]
Starting my first Axmol project. I'm working in xCode 15.0 on Mac 13.5.2.

Right off the bat I have the following 27 warnings all related to libjpeg.a. They are are all warnings in the app link phase (which means I have hunt through them to find an pertinent warning and errors related to my actual code.

```
- ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][59](jccolor-avx2.asm.o)', assuming: macOS
- ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][60](jccolor-sse2.asm.o)', assuming: macOS
- ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][61](jcgray-avx2.asm.o)', assuming: macOS
- ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][62](jcgray-sse2.asm.o)', assuming: macOS
- ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][63](jchuff-sse2.asm.o)', assuming: macOS
- ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][64](jcphuff-sse2.asm.o)', assuming: macOS
- ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][65](jcsample-avx2.asm.o)', assuming: macOS
- ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][66](jcsample-sse2.asm.o)', assuming: macOS
- ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][67](jdcolor-avx2.asm.o)', assuming: macOS
- ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][68](jdcolor-sse2.asm.o)', assuming: macOS
- ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][69](jdmerge-avx2.asm.o)', assuming: macOS
- ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][70](jdmerge-sse2.asm.o)', assuming: macOS
- ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][71](jdsample-avx2.asm.o)', assuming: macOS
- ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][72](jdsample-sse2.asm.o)', assuming: macOS
- ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][73](jfdctflt-sse.asm.o)', assuming: macOS
- ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][74](jfdctfst-sse2.asm.o)', assuming: macOS
- ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][75](jfdctint-avx2.asm.o)', assuming: macOS
- ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][76](jfdctint-sse2.asm.o)', assuming: macOS
- ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][77](jidctflt-sse2.asm.o)', assuming: macOS
- ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][78](jidctfst-sse2.asm.o)', assuming: macOS
- ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][79](jidctint-avx2.asm.o)', assuming: macOS
- ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][80](jidctint-sse2.asm.o)', assuming: macOS
- ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][81](jidctred-sse2.asm.o)', assuming: macOS
- ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][82](jquantf-sse2.asm.o)', assuming: macOS
- ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][83](jquanti-avx2.asm.o)', assuming: macOS
- ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][84](jquanti-sse2.asm.o)', assuming: macOS
- ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][85](jsimdcpu.asm.o)', assuming: macOS
```

**Comment by halx99** [comment]
Please note, current axmol not test on newer xcode version than 14.2, xcode_14.2 is recommended, please install it , refer to: https://global.simdsoft.com/notes/#install-xcode-manually

**Comment by halx99** [comment]
> Starting my first Axmol project. I'm working in xCode 15.0 on Mac 13.5.2.
> 
> Right off the bat I have the following 27 warnings all related to libjpeg.a. They are are all warnings in the app link phase (which means I have hunt through them to find an pertinent warning and errors related to my actual code.
> 
> ```
> - ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][59](jccolor-avx2.asm.o)', assuming: macOS
> - ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][60](jccolor-sse2.asm.o)', assuming: macOS
> - ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][61](jcgray-avx2.asm.o)', assuming: macOS
> - ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][62](jcgray-sse2.asm.o)', assuming: macOS
> - ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][63](jchuff-sse2.asm.o)', assuming: macOS
> - ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][64](jcphuff-sse2.asm.o)', assuming: macOS
> - ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][65](jcsample-avx2.asm.o)', assuming: macOS
> - ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][66](jcsample-sse2.asm.o)', assuming: macOS
> - ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][67](jdcolor-avx2.asm.o)', assuming: macOS
> - ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][68](jdcolor-sse2.asm.o)', assuming: macOS
> - ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][69](jdmerge-avx2.asm.o)', assuming: macOS
> - ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][70](jdmerge-sse2.asm.o)', assuming: macOS
> - ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][71](jdsample-avx2.asm.o)', assuming: macOS
> - ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][72](jdsample-sse2.asm.o)', assuming: macOS
> - ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][73](jfdctflt-sse.asm.o)', assuming: macOS
> - ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][74](jfdctfst-sse2.asm.o)', assuming: macOS
> - ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][75](jfdctint-avx2.asm.o)', assuming: macOS
> - ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][76](jfdctint-sse2.asm.o)', assuming: macOS
> - ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][77](jidctflt-sse2.asm.o)', assuming: macOS
> - ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][78](jidctfst-sse2.asm.o)', assuming: macOS
> - ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][79](jidctint-avx2.asm.o)', assuming: macOS
> - ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][80](jidctint-sse2.asm.o)', assuming: macOS
> - ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][81](jidctred-sse2.asm.o)', assuming: macOS
> - ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][82](jquantf-sse2.asm.o)', assuming: macOS
> - ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][83](jquanti-avx2.asm.o)', assuming: macOS
> - ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][84](jquanti-sse2.asm.o)', assuming: macOS
> - ld: warning: no platform load command found in '/Users/Plang/API/axmol-dev/thirdparty/jpeg-turbo/prebuilt/mac/libjpeg.a[x86_64][85](jsimdcpu.asm.o)', assuming: macOS
> ```

No build error, right?

**Comment by stevetranby** [comment]
Warning only, but probably something to improve eventually whenever someone gets around to it.

Game projects could add a linker flag `-ld_classic` to use the older linker (or style of linking) to suppress these warnings.

However, you would probably want to test the linking speed (time to link built target) to make sure you're not missing out on an improvement to build times or file size.

The folks over at the Blender project have a CMAKE update that uses the `ld_classic` linker flag, as well as discussion on the topic and issues.

Discussion:
https://projects.blender.org/blender/blender/pulls/110243

CMake Update: 
https://projects.blender.org/cmbasnett/bdk-blender/commit/37129735126aade1dff2cd185be5a9da0ac5453d

**Reply by halx99** [reply]
@stevetranby do you want create a PR to fix the warning?

**Reply by stevetranby** [reply]
I'm not sure if it's desired to have this as a default (like in the new game template)? If others find this discussion and complain about the warnings I think they can fairly easily add this linker flag to their project.

Also, ignoring the warnings for all will prevent knowing when the warning is fixed. Possibly once all the offending prebuilt libraries are re-built using the new linker? Anyway, I'm going to start porting a larger project over to Axmol soon, so I'll keep an eye on this an other things where I can help out.
